src/com/amazon/ion/benchmark/ParseAndCompareBenchmarkResults.java

```diff
@@ -57,110 +38,110 @@ public class ParseAndCompareBenchmarkResults {
-        if (score.getType().equals(IonType.FLOAT)) {
-            IonFloat scoreFloat = (IonFloat) score;
-            return scoreFloat.bigDecimalValue();
+    public static double[] preProcess(String benchmarkResult, String keyWord) throws Exception {
+        IonList rawData;
+        IonStruct benchmarkResultStruct = readHelper(benchmarkResult);
+        if (keyWord.equals(SPEED)) {
+            rawData = (IonList)((IonStruct) benchmarkResultStruct.get(PRIMARY_METRIC)).get(RAW_DATA);
         } else {
-            IonDecimal scoreDecimal = (IonDecimal) score;
-            return scoreDecimal.bigDecimalValue();
+            IonStruct secondaryMetricStruct = (IonStruct) benchmarkResultStruct.get(SECONDARY_METRIC);
+            rawData = (IonList) ((IonStruct) secondaryMetricStruct.get(keyWord)).get(RAW_DATA);
         }
+        double[] data = rawData.stream().flatMapToDouble(element -> toDouble((IonList)element)).toArray();
+        double[] processedData = removeOutliers(data);
+        return processedData;
     }
 
     /**
-     * Parse the benchmark result and extract the IonStruct which contains scores information.
-     *
-     * @param benchmarkResultFilePath is the file path of benchmark result.
-     * @param keyWord                 represents which aspect of scores are required to be extracted.
-     * @return an IonStruct which contains the scores information.
-     * @throws Exception if errors occur when create IonReader.
+     * Using Interquartile range method filtering out the outliers to make sure the raw data collection is normally distributed.
+     * @param data represents the raw data from the same benchmarking process but different iterations.
+     * @return the pre-processed data array without outlier.
      */
-    private static IonStruct parseScore(String benchmarkResultFilePath, String keyWord) throws Exception {
-        IonStruct scoreStruct;
-        IonStruct benchmarkResultStruct = readHelper(benchmarkResultFilePath);
-        if (keyWord.equals(SPEED)) {
-            scoreStruct = (IonStruct) benchmarkResultStruct.get(PRIMARY_METRIC);
-        } else {
-            IonStruct secondaryMetricStruct = (IonStruct) benchmarkResultStruct.get(SECONDARY_METRIC);
-            scoreStruct = (IonStruct) secondaryMetricStruct.get(keyWord);
+    public static double[] removeOutliers(double[] data) {
+        Percentile percentile = new Percentile();
+        double q1 = percentile.evaluate(data, 25.0);
+        double q3 = percentile.evaluate(data, 75.0);
+        double iqr = q3 - q1;
+
+        double lowerBound = q1 - 1.5 * iqr;
+        double upperBound = q3 + 1.5 * iqr;
+
+        List<Double> filteredData = new ArrayList<>();
+        for (double value : data) {
+            if (value >= lowerBound && value <= upperBound) {
+                filteredData.add(value);
+            }
         }
-        return scoreStruct;
+        return filteredData.stream().mapToDouble(d -> d).toArray();
```

Comment:
We can make this a lot more concise. (I'm only 95% sure the syntax of suggestion is correct, but I know the concept is sound.)
```java
return Arrays.stream(data).filter(d -> lowerBound <= d && d <= upperBound).toArray();
```

Reply:
Thanks for the suggestion.